plugin/src/main/java/io/jenkins/plugins/analysis/warnings/StyleLint.java

```diff
@@ -0,0 +1,48 @@
+package io.jenkins.plugins.analysis.warnings;
+
+import edu.hm.hafner.analysis.parser.checkstyle.CheckStyleParser;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.jenkinsci.Symbol;
+import hudson.Extension;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisModelParser;
+import io.jenkins.plugins.analysis.core.model.IconLabelProvider;
+import io.jenkins.plugins.analysis.core.model.StaticAnalysisLabelProvider;
+
+/**
+ * Provides a parser and customized messages for Stylelint. Delegates to {@link CheckStyleParser}.
+ *
+ * @author Alexander Brandes
+ */
+public class StyleLint extends AnalysisModelParser {
+    private static final long serialVersionUID = -7803000148685522042L;
+    private static final String ID = "stylelint";
+
+    /** Creates a new instance of {@link StyleLint}. */
+    @DataBoundConstructor
+    public StyleLint() {
+        super();
+        // empty constructor required for stapler
+    }
+
+    /** Descriptor for this static analysis tool. */
+    @Symbol("styleLint")
+    @Extension
+    public static class Descriptor extends AnalysisModelParserDescriptor {
+        /** Creates the descriptor instance. */
+        public Descriptor() {
+            super(ID);
+        }
+
+        @Override
+        public boolean canScanConsoleLog() {
+            return false;
+        }
+
+        @Override
+        public StaticAnalysisLabelProvider getLabelProvider() {
+            return new IconLabelProvider(getId(), getDisplayName(), getDescriptionProvider());
```

Comment:
Is there an SVG available as well?

Can you please check the license of the icon and add it to https://github.com/jenkinsci/warnings-ng-plugin/blob/master/plugin/src/main/webapp/icons/LICENSE.txt?

Reply:
> Is there an SVG available as well?

Yes, should I add it instead of the .png ones? I was under the impression they are needed.

Reply:
Not anymore. Some tools still have no SVG so both options will work. You need to use `SvgIconLabelProvider`.